benches/stash.rs

```diff
@@ -27,12 +27,76 @@ fn bench(b: &mut Bencher) {
 }
 
 #[bench]
-fn lookup(b: &mut Bencher) {
+fn bench_unchecked(b: &mut Bencher) {
+    let mut stash = Stash::new();
+    b.iter(|| {
+        let t1 = stash.put("something");
+        let t2 = stash.put("something");
+        let _ = unsafe{ stash.take_unchecked(t1) };
```

Comment:
You need to use `test::black_box` here to ensure this actually does something. Also, with the *current* API, you should be unwrapping.